web-src/src/components/GenerateButton.js

```diff
@@ -154,7 +156,7 @@ export function GenerateButton() {
         variant="cta"
         style="fill"
         onPress={handleGenerate}
-        isDisabled={generationInProgress}>
+        isDisabled={generationInProgress || syntaxError}>
```

Comment:
I'll probably leave this to @vtsaplin and @askayastha22 since they know the architecture of this app better. However, having a new shared Recoil state to control the availability of this button is not the best option, in my opinion. What if later there is another condition, such as a requirement to provide a target audience? Will another Recoil state need to be created for that?

I understand that the `GenerateButton` is already too coupled and it is impossible to reuse it in other components. However, it would be better to address the control of its availability at a minimum. The logic for this button's availability should be handled by the parent component, using the `isDisabled` prop, for example.